src/pages/experiments/[experimentId]/data-exploration/index.jsx

```diff
@@ -41,18 +42,42 @@ const ExplorationViewPage = ({
   const { windows, panel } = layout;
   const [selectedTab, setSelectedTab] = useState(panel);
   const [addMenuVisible, setAddMenuVisible] = useState(false);
-
+  const { method } = useSelector((state) => (
+    state.experimentSettings.processing?.configureEmbedding?.embeddingSettings || ' '
+  ));
   useEffect(() => {
     setSelectedTab(panel);
   }, [panel]);
 
+  useEffect(() => {
+    if (!method) {
+      dispatch(loadProcessingSettings(experimentId));
+    }
+  }, []);
+  const methodUppercase = method ? method.toUpperCase() : ' ';
+
+  useEffect(() => {
+    if (method && windows) {
+      dispatch(updateLayout({
+        ...windows,
+        first: {
+          ...windows.first,
+          first: {
+            ...windows.first.first,
+            first: `${methodUppercase} Embedding`,
+          },
+        },
+      }));
+    }
+  }, [method]);
```

Comment:
I think we can drastically simplify this approach. 
For a user to change the embedding method, they need to go back to data processing, to the calculate embedding plot and change one of the settings. Then this would trigger data processing pipeline and only after that they will be able to go to data exploration. This means that this use-effect will never be triggered -- change of the method is not possible without leaving data exploration. 

There is an object in this file called `categoryItems`. This object defines what each of the windows in data exploration is, where `key` denotes the title of the window. For your case, I think that all you need to do is to change the `key` for the first item under `Plots` to be equal to `${methodUppercase} Embedding`. And then you can delete this useEffect

Reply:
No this doesn't work, because the layout of the trees is in the store so this is what i firtly tried to do. It didn't work it was not rendering the windows properly, as it was a misscorrelation between the values in the initialState file. I can show you what I mean.